indico/modules/events/papers/controllers/display.py

```diff
@@ -214,3 +217,30 @@ def _process(self):
             reset_paper_state(self.paper)
             flash(_("The paper judgment has been reset"), 'success')
         return jsonify_data(html=render_paper_page(self.paper))
+
+
+class RHCallForPapers(RHPapersBase):
+    """Show the main CFP page"""
+
+    def _checkProtection(self):
+        if not session.user:
+            raise Forbidden
+        RHPapersBase._checkProtection(self)
+
+    def _checkParams(self, params):
+        RHPapersBase._checkParams(self, params)
+        self.contribs = get_contributions_with_user_as_submitter(self.event_new, session.user)
+        self.papers = get_contributions_with_paper_submitted_by_user(self.event_new, session.user)
+        self.contribs = list(set(self.contribs) - set(self.papers))
+
+    def _process(self):
+        return WPDisplayCallForPapers.render_template('display/call_for_papers.html', self._conf,
+                                                      event=self.event_new, contributions=self.contribs,
+                                                      papers=self.papers)
+
+
+class RHSelectContribution(RHCallForPapers):
+    """Select a contribution for which the user wants to submit a paper"""
+
+    def _process(self):
+        return WPDisplayCallForPapers.render_template('display/_select_contribution.html', contributions=self.contribs)
```

Comment:
Also, if this is rendered inside a dialog use `jsonify_template` instead of going through a WP